platform/darwin/src/darwin_conversion.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include <mbgl/util/logging.hpp>
+#include <mbgl/style/conversion.hpp>
+#include <mbgl/util/feature.hpp>
+#include <mbgl/util/optional.hpp>
+
+#include "MGLConversionValue.h"
+
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+
+inline bool isUndefined(const MGLConversionValue& value) {
+    return value.isNull();
+}
+
+inline bool isArray(const MGLConversionValue& value) {
+    return value.isArray();
+}
+
+inline bool isObject(const MGLConversionValue& value) {
+    return value.isObject();
+}
+
+inline std::size_t arrayLength(const MGLConversionValue& value) {
+    return value.getLength();;
+}
+
+inline mbgl::android::Value arrayMember(const MGLConversionValue& value, std::size_t i) {
+    return value.get(i);
+}
+
+inline optional<mbgl::android::Value> objectMember(const MGLConversionValue& value, const char* key) {
+    MGLConversionValue member = value.get(key);
+
+    if (!member.isNull()) {
+        return member;
+    } else {
+        return {};
+    }
+}
+
+template <class Fn>
+optional<Error> eachMember(const MGLConversionValue&, Fn&&) {
+    // TODO
+    // mbgl::Log::Warning(mbgl::Event::Android, "eachMember not implemented");
```

Comment:
Here, and in some other places, need to raise/log an error in whatever way is "right" for this SDK/project.